core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -544,7 +548,7 @@ public CompletableFuture<Optional<Throwable>> acknowledge(String memberId, List<
                             }
                             // Successfully updated the state of the offset.
                             updatedStates.add(updateResult);
-                            if (updateNextFetchOffset) {
+                            if (updateNextFetchOffset && updateResult.state != RecordState.ARCHIVED) {
```

Comment:
I don't like having this second condition here. Can we only have updateNextFetchOffset set to true in the cases where we want to update the next fetch offset?

Reply:
https://github.com/confluentinc/kafka/pull/1162#discussion_r1568909394